File: backend/code-challenges/java/src/main/java/LoanMatcher.java
import java.util.ArrayList;
import java.util.List;

public class LoanMatcher {
    private Customer customer;

    public LoanMatcher(Customer customer) {
        this.customer = customer;
    }

    public List<Loan> loans() {
        List<Loan> loans = new ArrayList<>();

        if (customer.income() <= 3000) {
            loans.add(new Loan("EMPRESTIMO_PESSOAL"));
        }

        return loans;
    }
}

File: backend/code-challenges/kotlin/src/main/kotlin/LoanMatcher.kt

import java.util.*

class LoanMatcher(private val customer: Customer) {

    fun loans(): List<Loan> {
        val loans: MutableList<Loan> = ArrayList()
        if (this.customer.income() <= 3000) {
            loans.add(Loan("EMPRESTIMO_PESSOAL"))
        }
        return loans
    }

}
